### PR TITLE
CXP-1284: Endpoint GET product mapping schema

### PR DESCRIPTION
**Pull request by masacc**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/src/Infrastructure/Controller/Public/GetCatalogProductMappingSchemaAction.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Messenger\Exception\ValidationFailedException;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class GetCatalogProductMappingSchemaAction
```
Comment (willy-ahva) [inline components/catalogs/back/src/Infrastructure/Controller/Public/GetCatalogProductMappingSchemaAction.php:26]:
```suggestion
final class GetProductMappingSchemaAction
```

File: components/catalogs/back/src/Infrastructure/Controller/Public/GetCatalogProductMappingSchemaAction.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        try {
+            $productMappingSchema = $this->queryBus->execute(new GetCatalogProductMappingSchemaQuery($catalogId));
+        } catch (ValidationFailedException $e) {
+            throw new ViolationHttpException($e->getViolations());
+        }
+
+        return new JsonResponse($productMappingSchema, Response::HTTP_OK);
+    }
+
+    private function getCatalog(string $id): Catalog
```
Comment (willy-ahva) [inline components/catalogs/back/src/Infrastructure/Controller/Public/GetCatalogProductMappingSchemaAction.php:55]: I wonder if we should put this in Trait too

**Review by willy-ahva — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Controller/Public/GetCatalogProductMappingSchemaAction.php (hunk 2 above)
Comment (masacc) [reply components/catalogs/back/src/Infrastructure/Controller/Public/GetCatalogProductMappingSchemaAction.php:55]: Something like `IsCatalogSubresourceTrait` can be cool I think

**Review by guillaumefer-akeneo — APPROVED** [review]

**Review by eoliveauakeneo — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Controller/Public/GetCatalogProductMappingSchemaAction.php (hunk 2 above)
Comment (masacc) [reply components/catalogs/back/src/Infrastructure/Controller/Public/GetCatalogProductMappingSchemaAction.php:55]: Do you want me to do this in this PR?

**Pull request merged**